coredns/endpointslice/map.go

```diff
@@ -142,7 +188,18 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 	for _, endpoint := range es.Endpoints {
 		var records []serviceimport.DNSRecord
 
-		for _, address := range endpoint.Addresses {
+		addresses := endpoint.Addresses
+		if localSlice != nil {
+			for _, localEndpoint := range localSlice.Endpoints {
+				if *localEndpoint.NodeName == *endpoint.NodeName &&
```

Comment:
The node name of `endpoint` wont be nil as it is set here https://github.com/submariner-io/lighthouse/blob/4a48ab0ef6769397760c6f268c04108c008a4e20/pkg/agent/controller/endpoint.go#L242 
The local endpoint could potentially have a nil node name but if that happens we not going to be able to tell if that endpoint is the one needed so should probably ignore it

Reply:
The `NodeName` pointer for `endpoint` is assigned from `address` which theoretically could be be nil. We just want to avoid a crash if somehow either is nil so, if we're not certain, then we should check for nil and perhaps log a warning.